beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/api/BeekeeperApiIntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+
+import com.expediagroup.beekeeper.api.BeekeeperApiApplication;
+import com.expediagroup.beekeeper.api.response.HousekeepingMetadataResponse;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.model.HousekeepingStatus;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+import com.expediagroup.beekeeper.integration.BeekeeperIntegrationTestBase;
+import com.expediagroup.beekeeper.integration.utils.BeekeeperApiTestClient;
+import com.expediagroup.beekeeper.integration.utils.RestResponsePage;
+
+public class BeekeeperApiIntegrationTest extends BeekeeperIntegrationTestBase {
+
+  public ObjectMapper geObjMapper() {
+    return new ObjectMapper()
+        .registerModule(new ParameterNamesModule())
+        .registerModule(new Jdk8Module())
+        .registerModule(new JavaTimeModule());
+  }
+
+  private static final Logger log = LoggerFactory.getLogger(BeekeeperApiIntegrationTest.class);
+
+  // APP CONTEXT AND TEST CLIENT
+  protected static ConfigurableApplicationContext context;
+  protected BeekeeperApiTestClient testClient;
+
+  protected final ObjectMapper mapper = geObjMapper();
+
+  @BeforeEach
+  public void beforeEach() {
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    int port = SocketUtils.findAvailableTcpPort();
+    String[] args = new String[] { "--server.port=" + port };
+    final String url = format("http://localhost:%d", port);
+    log.info("Starting to run Beekeeper API on: {} and args: {}", url, args);
+    context = SpringApplication.run(BeekeeperApiApplication.class, args);
+    testClient = new BeekeeperApiTestClient(url);
+  }
+
+  @AfterEach
+  public final void afterEach() {
+    log.info("Stopping Beekeeper API");
+    if (context != null) {
+      context.close();
+      context = null;
+    }
+  }
+
+  @Test
+  public void testGetMetadataWhenTableNotFound() throws SQLException, InterruptedException, IOException {
+    HousekeepingMetadata testMetadata1 = createHousekeepingMetadata("wrong_table",
+        "s3://some/path/event_date=2020-01-01/event_hour=0/event_type=A",
+        "event_date=2020-01-01/event_hour=0/event_type=A", LifecycleEventType.EXPIRED,
+        Duration.parse("P3D").toString());
+    HousekeepingMetadata testMetadata2 = createHousekeepingMetadata("wrong_table",
+        "s3://some/path/event_date=2020-01-01/event_hour=0/event_type=B",
+        "event_date=2020-01-01/event_hour=0/event_type=B", LifecycleEventType.EXPIRED,
+        Duration.parse("P3D").toString());
+    insertExpiredMetadata(testMetadata1);
+    insertExpiredMetadata(testMetadata2);
+
+    HttpResponse<String> response = testClient.getMetadata();
+    assertThat(response.statusCode()).isEqualTo(OK.value());
+    String body = response.body();
+    assertThrows(ValueInstantiationException.class,
```

Comment:
What is this? If you are getting an OK status then there should be a valid response from the API?

Reply:
Because when a table is not found, what is displayed is an empty list. Therefore there's not an error, the API still works, so the status is still OK

Reply:
As you should just read the value and the content length should be zero right? Not sure where the exception comes in

Reply:
Because it throws this exception: `Cannot construct instance of 
 com.expediagroup.beekeeper.integration.utils.RestResponsePage, problem: Page size must not be less than one!`, so if this exception is thrown it means the page size is 0 and no tables were found, which is what we want

Reply:
But that's a generic error jackson throws when it can't deserialize the object. You need to figure out a way to parse a the page as an empty. The main that's causing the error is `PageRequest.of(number, size)`. Size needs to be greater than 0. Maybe make a new object for empty page or modify this one.